app/controllers/DepartureTestMessagesController.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import connectors.DepartureConnector
+import javax.inject.Inject
+import models.DepartureId
+import models.MessageWithStatus
+import models.TestMessage
+import play.api.libs.json.JsValue
+import play.api.mvc.Action
+import play.api.mvc.ControllerComponents
+import repositories.DepartureRepository
+import uk.gov.hmrc.http.HttpErrorFunctions
+import uk.gov.hmrc.play.bootstrap.controller.BackendController
+import utils.Messages
+import utils.ResponseHelper
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.util.Random
+import scala.xml.NodeSeq
+
+class DepartureTestMessagesController @Inject()(cc: ControllerComponents, departureRepository: DepartureRepository, departureConnector: DepartureConnector)(
+  implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with HttpErrorFunctions
+    with ResponseHelper {
+
+  val messageTypes: Map[String, () => NodeSeq] = Map("IE928" -> Messages.generateIE928Message)
+
+  def injectEISResponse(departureId: DepartureId): Action[JsValue] = Action.async(parse.json) {
+    implicit request =>
+      withJsonBody[TestMessage] {
+        testMessage =>
+          testMessage.message.messageType match {
+            case "IE928" =>
```

Comment:
> If you are using MessageType, why haven't you included IE928 for the positive acknowledgement and directly referenced a PositiveAcknowledgement here. Then it could be `PositiveAcknowledgement.code` or `PositiveAcknowledement.rootNode` rather than a magic string.

I'll include PositiveAckowledgement but the magic string was used as the input is JSON

Reply:
> It may be better to do something like
> 
> `SupportedMessages.contains(testMessage.message.messageType)` It is also a candidate for moving to an action like with getting the departure. Then we can have the message process before we even access the db.

:thumbsup: